Gem/Code/Source/ShaderReloadSoakTestComponent.cpp

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzFramework/IO/LocalFileIO.h>
+
+#include <Atom/RPI.Public/View.h>
+
+#include <Atom/RPI.Reflect/Asset/AssetUtils.h>
+
+#include <SampleComponentManager.h>
+#include <SampleComponentConfig.h>
+
+#include "ShaderReloadSoakTestComponent.h"
+
+namespace AtomSampleViewer
+{
+    void ShaderReloadSoakTestComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class < ShaderReloadSoakTestComponent, AZ::Component>()
+                ->Version(0)
+                ;
+        }
+    }
+
+    ShaderReloadSoakTestComponent::ShaderReloadSoakTestComponent()
+    {
+    }
+
+    void ShaderReloadSoakTestComponent::InitTestDataFolders()
+    {
+        auto io = AZ::IO::LocalFileIO::GetInstance();
+
+        auto projectPath = AZ::Utils::GetProjectPath();
+        AZStd::string mainTestFolder;
+        AzFramework::StringFunc::Path::Join(projectPath.c_str(), "Shaders/ShaderReloadSoakTest/", mainTestFolder);
+        AzFramework::StringFunc::Path::Join(mainTestFolder.c_str(), "TestData/", m_testDataFolder);
+        if (!io->Exists(m_testDataFolder.c_str()))
+        {
+            AZ_Error("ShaderReloadSoakTestComponent", false, "Could not find source folder '%s'. This sample can only be used on dev platforms.", m_testDataFolder.c_str());
+            m_testDataFolder.clear();
+            return;
+        }
+
+        AzFramework::StringFunc::Path::Join(mainTestFolder.c_str(), "Temp/", m_tempSourceFolder);
+        if (!io->CreatePath(m_tempSourceFolder.c_str()))
+        {
+            AZ_Error("ShaderReloadSoakTestComponent", false, "Could not create temp folder '%s'.", m_tempSourceFolder.c_str());
+        }
+    }
+
+    void ShaderReloadSoakTestComponent::CopyTestFile(const char * originalName, const char * newName, bool replaceIfExists)
```

Comment:
Can any of this class be factored out with MaterialHotReloadTestComponent? Maybe form a common base class?

Reply:
I was thinking about all that, but I rather create a ticket for post reinvent.

Reply:
I expect my comments below about UI responsiveness and timing would be nicely alleviated by factoring out with MaterialHotReloadTestComponent